experimental/addons/azure-keyvault-csi/README.md

```diff
@@ -0,0 +1,56 @@
+
+# Azure Keyvault CSI Trait
+
+Uses the [Azure Key Vault Provider for Secrets Store CSI Driver](https://github.com/Azure/secrets-store-csi-driver-provider-azure)
+to provide secrets/keys/certs from an Azure Keyvault to the component using
+a CSI inline volume mounted at `/mnt/secrets-store`.
+
+The Azure Key Vault Provider and Secrets Store CSI Driver must already be 
+installed in the cluster - see [installation instructions](https://azure.github.io/secrets-store-csi-driver-provider-azure/docs/getting-started/installation/)
```

Comment:
Can we add the helm chart in the addon so that we can install this when we enable the addon?
You can use helm in addon like: https://github.com/kubevela/catalog/blob/master/addons/model-training/template.yaml#L9

Reply:
@FogDong I understand some logic in adding the helm install, but my concern is that there would need to be pass-through of all possible helm configuration values (see https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/charts/csi-secrets-store-provider-azure/README.md#configuration).

In addition, those with permission to install vela/addons may not have permission for the kube-system namespace on the cluster.

So the helm install during enabling the addon would need to be optional as well as fully configurable.

This is why it seems best to leave secrets store csi driver/azure keyvault provider as a separate action.

What do you think?

Reply:
We can make the values configurable in the addon, but if there're too many values to set, the addon will be hard to use. I remember that we're going to support passing values from `Values.yaml` file to the helm type component, we can support passing values from files when enabling the addon in the near future. cc @wangyikewxgm

For the permission thing, I'm not familiar with the azure provider, do they have to install in the `kube-system`?

IMO, I think it's better if we can make users choose whether to install the helm chart or not, but this feature will be supported after kubevela/kubevela#4401 is merged. The trait looks good to me and thank you for your contribution, I think we can merge this PR first and move forward after our feature is ready.